### PR TITLE
fix: address ux on pinned items drag-and-drop

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: #6278 

### Description:

Drag-and-drop enhancements:
- Add drag icon to corner of Resource Item that can be dragged. 
- Pass the drag handler props to the `DragIcon` instead
- Only show icon on hover
- Cursor should only be in _drag_ mode if the `DragIcon` is clicked

https://github.com/lightdash/lightdash/assets/7611706/33a7f6df-21ad-481a-8901-9ad054f72b83

Tooltip changes:
- only display tooltip on hover of the item `name`
- changed `Space`'s `Popover`->`Tooltip`, seems to be working fine!

https://github.com/lightdash/lightdash/assets/7611706/6a6ec937-cc2e-43f1-b613-8384369cdc02

NOTE: when viewing `All Spaces`, the drag and drop icon does not get displayed because draggability is disabled in that context.

### CODE REVIEW

File: packages/frontend/src/components/common/ResourceView/ResourceViewGrid/index.tsx [hunk 1]
```diff
@@ -32,6 +35,93 @@ type ResourceViewGridProps = ResourceViewGridCommonProps &
...
+};
+
+const DraggableItem: FC<DraggableItemProps> = ({
+    item,
+    index,
+    onAction,
+    projectUuid,
+    hasReorder,
+}) => {
+    const { hovered: isHovered, ref: hoverRef } = useHover<HTMLAnchorElement>();
```
Comment (joaoviana) [inline packages/frontend/src/components/common/ResourceView/ResourceViewGrid/index.tsx:53]: moved to its own component so I can use this hook ^

**Review by yasmin-e — APPROVED** [review]
looks good to me 🎉

**Comment by joaoviana** [comment]
> im not entirely sure about drag and drop icon and it's position. usually I'd expect it to appear on the left side and ::: kind of icon (that's the pattern we have) cc: @PriPatel
> 
> but code looks great! nice work @joaoviana !!!

found the icon @PriPatel requested, thanks for raising this again @IrakliJani !

Here is the updated version: 

https://github.com/lightdash/lightdash/assets/7611706/0a7cdb45-fd79-4d83-984d-4187a672b25c

**Review by PriPatel — COMMENTED** [review]
Looks great - thank you so much!

**Pull request merged**